helpers/AuditLogDynamoDbHelper.js

```diff
@@ -0,0 +1,50 @@
+const { DynamoDB } = require("aws-sdk");
+const isError = require("../utils/isError");
+
+class DynamoDbHelper {
+  constructor(config) {
+    const { region, endpoint, tableName } = config;
+    let dynamoDbOptions = { region };
+
+    if (endpoint) {
+      dynamoDbOptions = { ...dynamoDbOptions, endpoint };
+    }
+
+    const dynamoDb = new DynamoDB(dynamoDbOptions);
+
+    this.client = new DynamoDB.DocumentClient({
+      service: dynamoDb
+    });
+
+    this.tableName = tableName;
+  }
+
+  async getMessageByExternalCorrelationId(externalCorrelationId) {
+    const queryOptions = {
+      TableName: this.tableName,
+      IndexName: "externalCorrelationIdIndex",
+      KeyConditionExpression: "#keyName = :keyValue",
+      ExpressionAttributeValues: {
+        ":keyValue": externalCorrelationId
+      },
+      ExpressionAttributeNames: {
+        "#keyName": "externalCorrelationId"
+      },
+      ScanIndexForward: true,
+      Limit: 1
+    };
+
+    const result = await this.client
+      .query(queryOptions)
+      .promise()
+      .catch((error) => error);
+
+    if (isError(result)) {
+      throw result;
+    }
+
+    return result.Items[0];
```

Comment:
in case items is of length 0, is it worth throwing a controlled exception?

Reply:
or will that never happen in this query?

Reply:
This happens that the `Items` is empty but the value of `Items[0]` will be `undefined` which is a valid value for this function.